### ISSUE TITLE
PipelineCreateReplaceChangeSetAction does not allow for multiple input artifacts

### ISSUE THREAD

**Issue by seawatts** [awslabs/aws-cdk#1247]
From the AWS Pipeline console you can configure a change set action to have multiple input artifacts however, it is not exposed in the CDK. Under the hood it looks like the CDK implements a function called `addInputArtifact` however, it is a protected function.

**Comment by skinny85** [comment]
Hey Chris,

thanks for opening the issue. What's the behavior when specifying multiple input artifacts? Is the extra input used to specify parameter overrides? Or something else perhaps?

Thanks,
Adam

**Comment by seawatts** [comment]
Hey @skinny85 no problem, thanks for the quick reply.
That is actually exactly what I'm using this for. 

Currently I have a mono repo and each sub project I want to be able to deploy to it's own stack so that I can reduce the number of dependencies that are shipped for each project. My pipeline currently looks like source => build => (Update Step) 

Update Step has Stack1ChangeSet and Stack2ChangeSet each one get's built with a specific parameter template pointing to each specific build output from the build step. 

Does that all make sense?

**Comment by skinny85** [comment]
That's good, because that use case is actually supported :).

You would do something like this:

```ts
new cloudformation. PipelineCreateReplaceChangeSetAction(this, 'Stack1CreateChangeSetAction', {
  stackName: 'YourStack',
  changeSetName: 'YourChangeSet',
  templatePath: buildAction.outputArtifact.atPath('path/to/your/template.yml'),
  templateConfiguration: buildAction.outputArtifact.atPath('path/to/your/overrides/file.json'),
});
```

And similarly for `Stack2ChangeSet`.

Let me know if this solves your use case!

**Comment by skinny85** [comment]
Sorry, I'm not following. Can you show me your code?

**Comment by seawatts** [comment]
After talking with @skinny85 offline we have figured out that it is better to expose the PipelineCreateReplaceChangeSetAction to only include one input artifact and instead if we need resources from other build artifacts for cloud formation parameters we can just include those in our buildspec

**Comment by seawatts** [comment]
Hey, coming back around I ran into another use case where I need this feature.

In actual cloud formation this is possible
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-stages-actions.html#cfn-codepipeline-pipeline-stages-actions-inputartifacts

**Comment by eladb** [comment]
@skinny85 are you picking this up?

**Comment by skinny85** [comment]
@seawatts can you explain how are the extra input artifacts used in CloudFormation Actions? What do they do?

**Comment by skinny85** [comment]
OK, I understand how are the extra input artifacts used in CloudFormation Actions now. I'm working on fixing this.

**Comment by skinny85** [comment]
PR with the fix: #2020
